src/Qir/Runtime/lib/QIR/QirOutputHandling.cpp

```diff
@@ -0,0 +1,86 @@
+#include <cstdint>
+#include <sstream>
+#include "QirOutputHandling.hpp"
+#include "QirTypes.hpp"
+#include "QirRuntime.hpp"
+
+#define RECORD_TYPE "RESULT"
+#define RECORD_COLUMN_DELIMITER "\t"
+#define RECORD_PREFIX RECORD_TYPE RECORD_COLUMN_DELIMITER /* "RESULT" "\t" (== "RESULT\t") */
+
+
+static void PrintCStr(const char* cStr)
+{
+    QirString msg(cStr);
+    __quantum__rt__message(&msg);
```

Comment:
If I understand correctly, `__quantum_rt_message` appends the record prefix `INFO`, such that this function as written would result in program output of the form `INFO RESULT 0` rather than the desired `RESULT 0`?

Reply:
Looking at the `__quantum__rt__message()` [implementation](https://github.com/microsoft/qsharp-runtime/blob/8d305b9345f8f924abf6c5fdc2370a0a17883d06/src/Qir/Runtime/lib/QIR/Output.cpp#L11) and that the [`OutputStream` is just the `std::cout`](https://github.com/microsoft/qsharp-runtime/blob/8d305b9345f8f924abf6c5fdc2370a0a17883d06/src/Qir/Runtime/lib/QIR/OutputStream.cpp#L17-L22) (and that the `std::cout` is supposed to be [replaced](https://github.com/microsoft/qsharp-runtime/blob/8d305b9345f8f924abf6c5fdc2370a0a17883d06/src/Qir/Runtime/lib/QIR/OutputStream.cpp#L24-L43) for testing purposes only), I don't see any place where the mentioned `INFO` can be added.
What do I miss?

Reply:
I think we do eventually want the message intrinsic to show up as `INFO` records in the output, but that change hasn't been made. What will need when we go that way is a more generic output function, perhaps something like `__quantum__rt__record_output(char*, char*)` that takes both the tag and the content to be tagged, and then reimplement both the functions added here and message on top of those functions. I don't know if that's in-scope for this PR though...

Reply:
@cgranade, @swernli, can you please refer me to more details about that `INFO`? (a task/issue may be?) Looks like I really miss some story here.

Reply:
> I don't know if that's in-scope for this PR though...

Fair, yeah. I guess my suggestion, then, would be to anticipate that change by not calling `__quantum__rt_message` directly here, but renaming that implementation to some private thing like `__write_to_current_stream(char*)` and then making `__quantum__rt_message` a one-line function that just calls `__write_to_current_stream`. That could be about a five-line change, and would make it easier to modify `__quantum__rt_message` in the future if needed without having to chase down other dependencies that really are dependencies on the output stream in disguise.